packages/vite/src/node/plugins/css.ts

```diff
@@ -1055,6 +1058,19 @@ async function compileCSS(
   }
 }
 
+const lazyImportCache = new Map()
+function lazyImport<T>(name: string, imp: () => Promise<T>): T | Promise<T> {
+  const cached = lazyImportCache.get(name)
+  if (cached) return cached
+
+  const promise = imp().then((module) => {
+    lazyImportCache.set(name, module)
+    return module
+  })
+  lazyImportCache.set(name, promise)
+  return promise
+}
```

Comment:
Incredible how much we'll gain because of this, great find!

About the name, maybe `cachedImport` would be more clear here. And being general about the imp is interesting but we could directly call `import(name)` inside to avoid creating the `imp` function every time it is called?

Reply:
+1 for `cachedImport`

For `import(name)` I guess you need to pass it as literal so that bundler and typescript can infer it. So I think the current implementations LGTM.

Reply:
I don't think we need `import(name)` since Vite isn't being bundled, and I don't think Vite can be bundled now too due to some constant fs paths. We're already specifying deps dynamically with the `requireResolveFromRootWithFallback` util, so something like `cachedImport('postcss')` would be fine by me too.

Reply:
`postcss` is not bundled but `postcss-modules` and `postcss-import` is bundled.

Reply:
Ah right, let's keep it as-is then to make it simple 😄